src/mappings/poolCommitter.ts

```diff
@@ -177,17 +184,117 @@ export function executedCommitsForInterval(event: ExecutedCommitsForInterval): v
 
 			commit.upkeep = upkeepId
 			commit.isExecuted = true;
+			const typeRaw = commit.typeRaw;
 			// if the commit type is not a mint type
 			// the applicable fee is the burn fee
-			if (commit.typeRaw !== SHORT_MINT && commit.typeRaw !== LONG_MINT) {
+			if (typeRaw !== SHORT_MINT && typeRaw !== LONG_MINT) {
 				commit.applicableFee = burningFee;
 				commit.applicableFeeRaw = event.params.burningFee
 			}
+			commit.save();
+
+			if (!traders.includes(commit.trader)) {
+				traders.push(commit.trader)
+			}
+		}
+
+		// use library for price calcs
+		const poolSwapLibrary = PoolSwapLibrary.bind(poolSwapLibraryAddress);
+
+		// fetch token supplys
+		const longTokenInstance = ERC20.bind(poolEntity.longToken as Address);
+		const shortTokenInstance = ERC20.bind(poolEntity.shortToken as Address);
+		const longTokenSupply = longTokenInstance.totalSupply()
+		const shortTokenSupply = shortTokenInstance.totalSupply()
+		
+		// This event is emmitted after the supply has been changed so
+		// dont need to factor in burn amounts
+		const longPrice = floatingPointBytesToInt(poolSwapLibrary.getPrice(
+			poolInstance.longBalance(),
+			longTokenSupply
+		), poolEntity.quoteTokenDecimals);
+		const shortPrice = floatingPointBytesToInt(poolSwapLibrary.getPrice(
+			poolInstance.shortBalance(),
+			shortTokenSupply
+		), poolEntity.quoteTokenDecimals);
+
+		
+		for (let i = 0; i < traders.length; i++) {
+			let trader = traders[i]
+
+			const aggregateBalanceId = poolId+'-'+trader.toHexString();
+			let aggregateBalancesEntity = UserAggregateBalances.load(aggregateBalanceId);
+
+			if (!aggregateBalancesEntity) {
+				aggregateBalancesEntity = initUserAggregateBalance(poolId, trader)
+			}
+			const aggregateBalances = poolCommitterInstance.getAggregateBalance(trader as Address);
```

Comment:
I believe this is the only contract call we need to add with this current change, however since we have long/short prices and complete commit info (amount in, commit type and applicable fee converted to integer), we should be able to calculate the change in user holdings for the current interval without needing to do any on-chain calls.

If we decide to do this, we would probably want to move some of this logic into the loop which goes through the commits above.

Reply:
yeah we would have to sum all the commit types and then subtract or add from previous stored aggregate balances. I just thought it would be easier to use whats in the contracts

Reply:
yeah the only thing I don't like about that is that it scales badly as the number of unique traders increases, I'm writing up a notion doc with formulas to calculate commit outcomes for Dan to use in the commit modal summary, should be able to use the same calcs here too